mozetl/engagement/churn/job.py

```diff
@@ -139,15 +170,30 @@ def extract(main_summary, new_profile, start_ds, period, slack, is_sampled):
     if is_sampled:
         predicates.append((F.col("sample_id") == "57"))
 
-    return (
+    extract_ms = (
         main_summary
         .where(reduce(operator.__and__, predicates))
         .select(SOURCE_COLUMNS)
-        .union(
-            clean_new_profile(new_profile)
-            .where(reduce(operator.__and__, predicates))
-            .select(SOURCE_COLUMNS)
-        )
+    )
+
+    np = clean_new_profile(new_profile)
+    extract_np = (
+        np
+        .where(reduce(operator.__and__, predicates))
+        .select(SOURCE_COLUMNS)
+    )
+
+    np_attribution = (
+        np
+        .where("attribution is not null")
```

Comment:
This filter is now done twice, here and in line 134.

Reply:
I'm removing this from line 134, it makes sense to push down the filter.